From 70e4afbea80df1de0084c5ce6119ed1b3465a2d3 Mon Sep 17 00:00:00 2001
From: Wen He <wen.he_1@nxp.com>
Date: Thu, 9 May 2019 11:31:01 +0800
Subject: [PATCH 506/741] drm/imx/hdp: Add driver for LS1028AQDS eDP panel
 support

commit 78fbe7986af1e262185bdfeb56b8871ec2ab62da from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

This patch add the Embedded DisplayPort Panel driver support
on LS1028AQDS board.

Signed-off-by: Wen He <wen.he_1@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-dp.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-dp.c b/drivers/gpu/drm/imx/hdp/imx-dp.c
index 73f2a3194b59..2864e96dd7b4 100644
--- a/drivers/gpu/drm/imx/hdp/imx-dp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-dp.c
@@ -284,6 +284,7 @@ void dp_mode_set(state_struct *state,
 	u8 num_lanes = 4;
 	u8 ssc = 0;
 	u8 scrambler = 1;
+	u8 edp_enable = 0;
 	/* Max voltage swing */
 	u8 max_vswing = 3;
 	u8 force_max_vswing = 0;
@@ -342,6 +343,8 @@ void dp_mode_set(state_struct *state,
 		default:
 			sym_rate = RATE_1_6;
 		}
+
+		edp_enable = 1;
 	} else {
 		link_rate = max_link_rate;
 
@@ -361,7 +364,8 @@ void dp_mode_set(state_struct *state,
 
 	ret = CDN_API_DPTX_SetHostCap_blocking(state,
 		link_rate,
-		(num_lanes & 0x7) | ((ssc & 1) << 3) | ((scrambler & 1) << 4),
+		(num_lanes & 0x7) | ((ssc & 1) << 3) | ((scrambler & 1) << 4 |
+		((edp_enable) & 1) << 5),
 		(max_vswing & 0x3) | ((force_max_vswing & 1) << 4),
 		(max_preemph & 0x3) | ((force_max_preemph & 1) << 4),
 		supp_test_patterns,
-- 
2.17.1

